internal/proxy/env.go

```diff
@@ -9,15 +9,29 @@ import (
 
 	log "github.com/sirupsen/logrus"
 	"github.com/ubuntu/decorate"
+	"golang.org/x/exp/slices"
 )
 
+// unsupportedEnvProtocols lists protocols that are not supported by the environment proxy.
+var unsupportedEnvProtocols = []protocol{protocolAuto}
+
 // envString formats a proxy setting to be environment variable compliant.
 func (p setting) envString() string {
+	if slices.Contains(unsupportedEnvProtocols, p.protocol) {
+		return ""
+	}
```

Comment:
In the end, it's up to you. I feel like, when it comes to user feedback, it can never be too much. I agree that logging it 5 times is a bit annoying, but for this kind of errors I feel like it's important to be explicit about it, no? Even though we already document it, it doesn't hurt to log it back to the users should them try to set an unsupported protocol (specially since they differ depending on the proxy setting).
Nevertheless, the final decision is yours.